### ISSUE TITLE
Support .NET Standard new appsettings

### ISSUE THREAD

**Issue by 304NotModified** [NLog/NLog#2334]

**Comment by ericnewton76** [comment]
wow, just found out about netstandard appsettings because of this issue.  looks interesting.

how would you guys end up supporting it?  #if statements or something?

you could almost make the argument to extract the netfranework configuration out and either configuration pluggable via (even poor-mans) DI:

ALthough I like the "default configuration handling" that has benefited me once or twice, I dont think it would be too much issue to require establishing NLog configuration intialization somewhere in an App_Start context. and possibly a (blocking) Func pointer that could be invoked if any LogMessage called and configuration not available yet.

**Comment by snakefoot** [comment]
@ericnewton76 Think we need this in place before we can load the entire configuration from a json-file: #1588

**Comment by 304NotModified** [comment]
You're right @snakefoot. This issue is for ${appsetting}

**Comment by snakefoot** [comment]
@304NotModified I guess this can be removed from NLog 4.6, and be closed as resolved by  #245 (And will soon be merged into the NLog.Extension.Logging). When #248 starts rolling.

**Comment by 304NotModified** [comment]
Yes, thanks!

**Comment by 304NotModified** [comment]
> NLog.Extension.Logging when #248 starts rolling).

you mean https://github.com/NLog/NLog.Extensions.Logging/pull/248, right?

**Comment by snakefoot** [comment]
> @304NotModified you mean NLog/NLog.Extensions.Logging#248, right?

Yes. Have now fixed my link
